pypdf/_text_extraction/_layout_mode/_xform_stack.py

```diff
@@ -0,0 +1,170 @@
+"""manage the PDF transform stack during "layout" mode text extraction"""
+
+from collections import ChainMap, Counter
+from typing import Any, Dict, List, MutableMapping, Union
+from typing import ChainMap as ChainMapType
+from typing import Counter as CounterType
+
+from .. import mult
+from ._fonts import Font, TextStateParams
+
+XformStackChainMapType = ChainMapType[Union[int, str], Union[float, bool]]
+XformStackDictType = MutableMapping[Union[int, str], Union[float, bool]]
+
+class XformStack:
+    """
+    A cm/tm transformation matrix manager
+
+    Attributes:
+        xfrm_stack (ChainMap): ChainMap of cm/tm transformation matrices
+        q_queue (Counter[int]): Counter of q operators
+        q_depth (List[int]): list of q operator nesting levels
+        Tc (float): character spacing
+        Tw (float): word spacing
+        Tz (int): horizontal scaling
+        TL (float): leading
+        Ts (float): text rise
+        font (Font): font object
+        font_size (int | float): font size
+
+    Methods:
+        raw_xform: only a/b/c/d/e/f matrix params
+        new_xform: a/b/c/d/e/f matrix params + 'is_text' and 'is_render' keys
+        reset_tm: clear all xforms from chainmap having is_text==True
+        reset_trm: clear all xforms from chainmap having is_render==True
+        remove_q: rewind to stack prior state after closing a 'q' with internal 'cm' ops
+        add_q: add another level to q_queue
+        add_cm: concatenate an additional transform matrix
+        add_tm: append a text transform matrix
+        add_trm: append a text rendering transform matrix
+        text_state_params: current text state parameters
+        effective_xform: the current effective transform account for both cm and text xforms
+        flip_vertical: True if page is upside down
+        xmaps: internal ChainMap 'maps' property
```

Comment:
I think I'd rather have that just in the docstrings of the methods themselves.

```suggestion
```

Reply:
Makes sense. Adjustment made. This was a Copilot suggestion I didn't hate even though it's not my normal practice. ;)